### DISCUSSION TITLE
Poetry cannot install Emmett package

### DISCUSSION THREAD

**Discussion by zvolsky** [python-poetry/poetry#3681] [Q&A] [answered]
Please could somebody take a look at the Emmett package, if it is not correct or if Poetry has a bug?

Py 3.9, clear poetry virtualenvironment (poetry init + poetry shell)

This run well: `pip install emmett`.
But Poetry will fail with RootCause. In fact I am not sure. The code bellow looks like DependencyCause, however I have tracked RootCause in my debugging attempt.

```
(emmett-HhKsSRhV-py3.9) mirek@debian-mirek:~/emt/emmett/emmett$ poetry add emmett
Using version ^2.1.4 for emmett

Updating dependencies
Resolving dependencies... (0.0s)

AssertionError

at ~/.poetry/lib/poetry/mixology/incompatibility.py:111 in str
107│ )
108│
109│ def str(self):
110│ if isinstance(self._cause, DependencyCause):
→ 111│ assert len(self._terms) == 2
112│
113│ depender = self._terms[0]
114│ dependee = self._terms[1]
115│ assert depender.is_positive()

```

**Comment by finswimmer** [comment]
Hello @zvolsky,

no problem here. Which version of poetry are you using?

fin swimmer

**Reply by zvolsky** [reply]
I am sorry for this.
This was completely my mistake.
I wanted test an emmett web framework for the 1st time and I made the directory called emmett/ for my work.
This is a reason of failure.
Of course Poetry error message could be more friendly to quickly understand what happens.

**Comment by zvolsky** [comment, marked as answer]
This was mistake.
Same name for folder+project (emmett/) and installed dependency (emmett).
